### PR TITLE
Write how-to for Cloudscale installation

### PR DESCRIPTION
**Pull request by srueg**
See https://github.com/appuio/terraform-openshift4-cloudscale/pull/1

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+export BASE_DOMAIN=ocp4-poc.appuio-beta.ch
+export PULL_SECRET=<redhat-pull-secret> # From https://cloud.redhat.com/openshift/install/pull-secret
+----
+
+. Create S3 buckets
++
+[source,console]
+----
+# Use already exiting bucket user
+curl -H "Authorization: Bearer ${CLOUDSCALE_TOKEN}" \
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:48]: Do I have to do something with the output? I guess, wetter I use an existing user user or create one, the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` variables must be set. I can not see the result of this command being used anywhere (e.g. assigned to the variable `response`).

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+----
+
+. Prepare `install-config.yaml`
++
+[source,console]
+----
+mkdir ${CLUSTER_NAME}
+
+cat > "${CLUSTER_NAME}/install-config.yaml" <<EOF
+apiVersion: v1
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:78]: If I am not mistaken, most of that config is default anyway. We have this now all over the place. Can we just create configs with the bare minimum? Does this come from the OpenShift documentation?

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+. Wait for bootstrap to complete
++
+[source,console]
+----
+openshift-install --dir ${CLUSTER_NAME} \
+  wait-for bootstrap-complete
+----
+
+. Remove bootstrap node
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:158]: Can be omitted. The next step will run that target anyway.

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc [hunk 4]
```diff
@@ -37,6 +37,7 @@ It also will become a subdomain to the base domain.
 `<service account key>`:: See <<how-tos/gcp/project.adoc#service-account-key, Service Account Key>>
 `<region>`:: The GCP region to place the cluster. Default to `europe-west6`
 
+.Install cluster
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/install/gcp.adoc:40]: This is a title for the next paragraph which is a list item. Plus: why is this within that PR? It is not related.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+export AWS_ACCESS_KEY_ID=$(echo $response | jq -r .keys[0].access_key)
+export AWS_SECRET_ACCESS_KEY=$(echo $response | jq -r .keys[0].secret_key)
+
+mc config host add \
+  s3 https://objects.rma.cloudscale.ch ${AWS_ACCESS_KEY_ID} ${AWS_SECRET_ACCESS_KEY}
+
+mc mb --ignore-existing\
+  "s3/${CLUSTER_NAME}-bootstrap-ignition" \
+  "s3/${CLUSTER_NAME}-tf-state"
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:65]: With AKS and EKS we use one single S3 bucked (on AWS) to store the states. We then use workspaces to switch between states of different environments.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc (hunk 2 above)
Comment (srueg) [reply docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:78]: Yes, this is the bare minimum required for an installation.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc (hunk 3 above)
Comment (srueg) [reply docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:158]: I wanted this to be a separate step to make sure the bootstrap node is gone before adding nodes.

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc (hunk 4 above)
Comment (srueg) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:40]: Currently the instructions are under the "Required input" paragraph: https://openshift.docs.vshn.ch/oc4/how-tos/install/gcp.html

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc (hunk 2 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:71]: I remember that in last fall, we have created clusters with only a hand full of settings. Al those master, worker and networking setting could just be ommited.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc (hunk 3 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:151]: Fair.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 6]
```diff
@@ -0,0 +1,204 @@
+= Installation on Cloudscale
+
+[abstract]
+--
+Steps to install an OpenShift 4 cluster on https://cloudscale.ch[Cloudscale].
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:5]:
```suggestion
Steps to install an OpenShift 4 cluster on https://cloudscale.ch[cloudscale.ch].
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+These steps follow the https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal.html[Installing a cluster on bare metal] docs to set up a user provisioned installation (UPI).
+https://www.terraform.io[Terraform] is used to provision the cloud infrastructure.
+--
+
+== Prerequisites
+* Cloudscale API token
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:12]:
```suggestion
* cloudscale.ch API token
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+* `terraform`
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:13]: We might need to mention the Terraform version, either here or in the Terraform repository.

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+* `mc` https://docs.min.io/docs/minio-client-quickstart-guide.html[Minio client]
+* `jq`
+* Clone of the https://github.com/appuio/terraform-openshift4-cloudscale[terraform-openshift4-cloudscale] repository
+
+
+== Cluster Installation
+
+. Create a new Syn cluster: https://control.vshn.net/syn/lieutenantclusters
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:21]:
```suggestion
. Register the new OpenShift 4 cluster in Lieutenant: https://control.vshn.net/syn/lieutenantclusters
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
++
+[source,console]
+----
+oc create secret generic image-registry-private-configuration-user \
+--namespace openshift-image-registry \
+--from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=${AWS_ACCESS_KEY_ID} \
+--from-literal=REGISTRY_STORAGE_S3_SECRETKEY=${AWS_SECRET_ACCESS_KEY}
+----
+
+. Synfection
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:197]:
```suggestion
. Make the cluster Project Syn enabled
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
++
+Synfect the cluster according to https://wiki.vshn.net/x/ngMBCg
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:199]:
```suggestion
Install Steward on the cluster according to https://wiki.vshn.net/x/ngMBCg
```

File: docs/modules/ROOT/partials/nav.adoc [hunk 12]
```diff
@@ -1,19 +1,24 @@
 * xref:index.adoc[Home]
 * Tutorials
+
 * How-to guides
 ** Authentication
 *** xref:how-tos/authentication/sudo.adoc[Sudo]
+** Cloudscale
```
Comment (tobru) [inline docs/modules/ROOT/partials/nav.adoc:7]:
```suggestion
** cloudscale.ch
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc [hunk 13]
```diff
@@ -0,0 +1,204 @@
+= Installation on Cloudscale
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:1]:
```suggestion
= Installation on cloudscale.ch
```

File: docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc (hunk 8 above)
Comment (srueg) [reply docs/modules/ROOT/pages/how-tos/install/cloudscale.adoc:13]: It is already enforced here: https://github.com/appuio/terraform-openshift4-cloudscale/blob/init/provider.tf#L2

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc (hunk 4 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:40]: I see what the issues is. When rendered, one can not easily understand the scoping of the block titles. To solve this, we need to replace the the block titles with real section titles. I would prefer to do so within a PR on its own.

Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc]: See https://github.com/appuio/openshift4-docs/pull/22.

**Pull request merged**